spring-cloud-azure-eventhub-stream-binder/README.md

```diff
@@ -1,6 +1,33 @@
 # Spring Cloud Azure Event Hub Stream Binder
 
-The project provides **Spring Cloud Stream Binder for Azure Event Hub** which allows you to build your message-driven applications using Spring programming model and [Azure Event Hub](https://azure.microsoft.com/en-us/services/event-hubs/) service. 
+The project provides **Spring Cloud Stream Binder for Azure Event Hub** which allows you to build message-driven 
+microservice using **Spring Cloud Stream** based on [Azure Event Hub](https://azure.microsoft.com/en-us/services/event-hubs/) service.
+
+## EventHub Binder Overview
+
+The Spring Cloud Stream Binder for Azure Event Hub provides the binding implementation for the Spring Cloud Stream.
+This implementation uses Spring Integration Event Hub Channel Adapters at its foundation. From design's perspective, 
+Event Hub is similar as Kafka. Also, Event Hub could be accessed via Kafka API. If your project has tight dependency 
+on Kafka API, you can try [Event Hub with Kafka API Sample](../spring-cloud-azure-samples/spring-cloud-azure-eventhub-kafka-sample/)
+
+### Consumer Group
+
+Event Hub provides similar support of consumer group as Apache Kafka, but with slight different logic. While Kafka 
+stores all committed offsets in the broker, you have to store offsets of event hub messages 
+being processed manually. Event Hub SDK provide the function to store such offsets inside Azure Storage Account. So 
+that's why you have to fill `spring.cloud.stream.eventhub.checkpointStorageAccount`.
+
+### Partitioning Support
+
+Event Hub provides similar concept of physical partition as Kafka. But unlike Kafka's auto rebalancing between 
+consumers and partitions, Event Hub provides a kind of preemptive mode. Storage account acts as lease to 
+determine which partition is owned by which consumer. When a new consumer starts, it will try to steal some partitions 
+from most heavy-loaded consumer to achieve workload balancing.
+
+### Checkpoint Mode
+
+Currently, event hub binder considers one message as successfully processed once received and sent to downstream
+ channel. In the future, we might support manual checkpoint mode.
```

Comment:
Suggest we put "Manual checkpoint mode is now supported yet." in order not to make a commitment here.